packages/block-editor/src/components/block-list/block.js

```diff
@@ -306,7 +306,7 @@ const applyWithDispatch = withDispatch( ( dispatch, ownProps, { select } ) => {
 					clientId
 				);
 				if ( previousBlockClientId ) {
-					mergeBlocks( previousBlockClientId, clientId );
+					mergeBlocks( previousBlockClientId, clientId, -1 );
```

Comment:
added an `initialPosition` param to denote the direction of merge when backspacing to fix a bug in  [useFocusFirstElement](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-list/use-block-props/use-focus-first-element.js#L99) hook where it was not detecting reverse merging via backspace due to initialPosition not being set